source/elements/oneDAL/source/data_management/data_sources.rst

```diff
@@ -1,7 +1,117 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 .. _data-sources:
 
 ============
 Data Sources
 ============
 
-TBD
+This section describes the types related to the :txtref:`data-source` concept.
+
```

Comment:
What's about high-level requirements for data_source implementations?

Reply:
I'm not sure that there are such requirements. It is very much like the algorithm descriptor - different data source will not have anything in common probably.